json-endpoints/class.wpcom-json-api-site-user-endpoint.php

```diff
@@ -97,13 +96,23 @@ public function update_user( $user_id, $blog_id ) {
 				}
 			}
 		}
+
 		if ( isset( $input[ 'roles' ] ) ) {
+			// For now, we only use the first role in the array.
 			if ( is_array( $input['roles'] ) ) {
 				$user['role'] = $input['roles'][0];
-			} else {
+			} else if ( is_string( $input['roles'] ) ) {
 				$user['role'] = $input['roles'];
+			} else {
+				return new WP_Error( 'invalid_input', __( 'The roles property must be a string or an array.' ), 400 );
```

Comment:
Yes, we usually translate error strings, thanks for pointing out that there are missing textdomains!